core/src/test/scala/com/databricks/labs/remorph/transpilers/SnowflakeToDatabricksTranspilerTest.scala

```diff
@@ -268,11 +268,6 @@ class SnowflakeToDatabricksTranspilerTest extends AnyWordSpec with TranspilerTes
         "SELECT column_a as alias_a FROM table_a where column_a = '123';"
     }
 
-    "transpile LCA not replacing aliases" in {
```

Comment:
The expected output query was wrong (failed on Databricks with `[UNRESOLVED_COLUMN.WITH_SUGGESTION] A column, variable, or function parameter with name `alias_a` cannot be resolved.`)

Reply:
Does this mean that the transformation was incorrect and we were generating incorrect output, or that the expected output was incorrect and we had targeted producing this incorrect output. It would appear that the test output was not verified against Databricks in either case though. We need to make sure that everyone does this or we create tests that seem to pass but in fact generate bad output (which we have told the tests to expect).

Reply:
Explained as the test being incorrect and redundant.

Reply:
For the record, the expected output is correct, see attached screenshot:
<img width="481" alt="Screenshot 2024-11-27 at 12 17 42 PM" src="https://github.com/user-attachments/assets/5b4836ff-a38b-4920-9467-20a724dfd57a">
We should add the test back in, as it is definitely a valid twisted way someone can write things.

Reply:
Actually the test should read as follows:
  ```
  "transpile LCA not replacing aliases" in {
      "SELECT '123' as alias_a FROM table_a where alias_a = '123';" transpilesTo
        "SELECT '123' as alias_a FROM table_a where '123' = '123';"
    }
```
@gueniai your example is also valid but 'WITH TABLE...' creates a context which is not available in this test

Reply:
@gueniai your example above works because of homonymy, the following won't work:

```sql
WITH 
  table_a(col1) AS (SELECT '123' as foo)
SELECT table_a.col1 as alias_a FROM table_a where alias_a = '123'
```

Reply:
> Actually the test should read as follows:
> 
> ```
> "transpile LCA not replacing aliases" in {
>     "SELECT '123' as alias_a FROM table_a where alias_a = '123';" transpilesTo
>       "SELECT '123' as alias_a FROM table_a where '123' = '123';"
>   }
> ```
That is right, but it felt a bit redundant with the test above so I decided to delete it altogether

Reply:
The one above doesn't use a literal as alias expression, so I'd say it's not redundant